bootstrap/src/main/java/com/facebook/airlift/bootstrap/Bootstrap.java

```diff
@@ -326,24 +327,26 @@ static SortedMap<String, String> replaceWithEnvironmentVariables(
     {
         SortedMap<String, String> results = new TreeMap<>();
         properties.forEach((key, value) -> {
-            Matcher result = ENV_PATTERN.matcher(value);
-            if (result.matches()) {
-                String variableName = result.group(1);
-                String envValue = environment.get(variableName);
-
-                if (envValue == null) {
-                    String errorMessage = String.format(
-                            "Configuration property `%s` references `%s`, an undefiled environment variable.",
-                            key,
-                            variableName);
-                    onError.accept(key, errorMessage);
+            // TODO: Replace with Matcher.replaceAll(Function) or replace StringBuffer with StringBuilder after upgrade to Java 9+
+            StringBuffer result = new StringBuffer();
+            ImmutableSet<String> undefined = ImmutableSet.of();
```

Comment:
```suggestion
            ImmutableSet.Builder<String> undefined = ImmutableSet.builder();
```

Reply:
Suggested change make sense, because in general it is a correct way to use immutable collections. However, current implementation is based on the assumption that the number of environment variables in real configuration should be near zero. So, creating empty collection by default should be a better option to performance and garbage collection. What do you think?
As alternative, `ImmutableSet.<String>builderWithExpectedSize(0)` can probably be used here.